controllers/controll_plane_test.go

```diff
@@ -167,6 +173,57 @@ var _ = Describe("minimal", func() {
 
 			}, existingClusterTimeout, existingClusterInterval).Should(Succeed())
 
+			serverName := common.OrdinalFQDNS(crd.Name, defaultNamespace, 0)
+			By("setting up operator identity on http client")
+			httpClient := http.Client{
+				Transport: http.DefaultTransport,
+				// A timeout less than 3 seconds may cause connection issues when
+				// the server requires to change the chiper.
+				Timeout: time.Second * 3,
+			}
+
+			httpClientTransport := httpClient.Transport.(*http.Transport)
+			httpClientTransport.TLSClientConfig = &tls.Config{
+				ServerName:         serverName,
+				InsecureSkipVerify: false,
+			}
+			httpClientTransport.TLSClientConfig.GetClientCertificate =
+				func(cri *tls.CertificateRequestInfo) (*tls.Certificate, error) {
+					return common.GetOperatorClientCertificate(k8sClient, cri)
+				}
+
+			if rootCas, err := common.GetRootCAs(k8sClient); err == nil {
+				httpClientTransport.TLSClientConfig.RootCAs = rootCas
+			}
+
+			By("Checking metrics with mtls are visible")
+			Eventually(func(g Gomega) {
+
+				resp, err := httpClient.Get("https://" + serverName + ":8888/metrics")
+				if verbose {
+					fmt.Printf("Resp form metrics get resp: %v, error: %v\n", resp, err)
+				}
+				g.Expect(err).Should(Succeed())
```

Comment:
Should an HTTP request with another certificate fail?

Reply:
the next step could be to allow the the rbac to be extended, possibly by providing keys to config in some control plane override secret